libs/classpath_resolver/src/resolver.rs

```diff
@@ -0,0 +1,126 @@
+use std::{
+    fs,
+    io::{self, Write},
+    path::{Path, PathBuf},
+    process::Command,
+    result,
+};
+
+#[derive(Debug)]
+pub enum WorkspaceType {
+    Maven,
+    Gradle,
+}
+
+pub struct ClasspathResolver {
+    workspace_type: Option<WorkspaceType>,
+    workspace_root: PathBuf,
+    classpath: Vec<PathBuf>,
+}
+
+impl ClasspathResolver {
+    pub fn new_with_type(
+        workspace_root: PathBuf,
+        workspace_type: Option<WorkspaceType>,
+    ) -> ClasspathResolver {
+        let classpath = match workspace_type {
+            Some(WorkspaceType::Maven) => resolve_maven_classpath(&workspace_root),
+            Some(WorkspaceType::Gradle) => resolve_gradle_classpath(&workspace_root),
+            None => Ok(vec![]),
+        };
+
+        ClasspathResolver {
+            workspace_type,
+            workspace_root,
+            classpath: classpath.unwrap_or(vec![]),
+        }
+    }
+
+    pub fn new(workspace_root: PathBuf) -> ClasspathResolver {
+        let workspace_type = detect_workspace_type(&workspace_root);
+
+        ClasspathResolver::new_with_type(workspace_root, workspace_type)
+    }
+}
+
+fn detect_workspace_type(workspace_root: &PathBuf) -> Option<WorkspaceType> {
+    if workspace_root.join("pom.xml").exists() {
+        return Some(WorkspaceType::Maven);
+    } else if workspace_root.join("build.gradle").exists()
+        || workspace_root.join("build.gradle.kts").exists()
+    {
+        return Some(WorkspaceType::Gradle);
+    }
+
+    None
+}
+
+fn resolve_maven_classpath(workspace_root: impl AsRef<Path>) -> Result<Vec<PathBuf>> {
+    // Resolve the classpath for a Maven workspace
+    Ok(vec![])
+}
+
+fn resolve_gradle_classpath(workspace_root: impl AsRef<Path>) -> Result<Vec<PathBuf>> {
+    let init_script = get_init_script_location();
+    let args: [&str; 3] = ["-I", &init_script, "classpaths"];
+
+    let output = run_shell(
+        gradle_command(),
+        &args,
+        workspace_root,
+    )?;
+
+    let classpaths = output
+        .lines()
+        .collect::<Vec<&str>>()
+        .into_iter()
+        .filter(|entry| entry.starts_with("gradle-classpath-resolver "))
+        .map(|entry| {
+            let path = entry.replace("gradle-classpath-resolver ", "");
+            PathBuf::from(path)
+        })
+        .collect::<Vec<PathBuf>>();
+
+    Ok(classpaths)
+}
+
+fn get_init_script_location() -> String {
+    let script = PathBuf::from("classpath.gradle");
+
+    fs::canonicalize(script)
+        .unwrap()
+        .to_str()
+        .unwrap()
+        .to_string()
```

Comment:
The unwraps here can panic. We don't ever want to panic

Reply:
For this one. The file is always in the repo so logically it is impossible for it to panic